components/aws/sagemaker/tests/integration_tests/scripts/run_integration_tests

```diff
@@ -78,7 +80,8 @@ function cleanup() {
   set +e
 
   cleanup_kfp
-  delete_generated_role
+  delete_assumed_role
```

Comment:
if we don't specify the "ASSUMED_ROLE_NAME" no new role it created  
but we r still deleting it here .. which will delete pre-existing role ?

Reply:
If we don't specify that role name (which is the default option) then it creates one automatically. This method ensures it is cleaned up correctly.

Reply:
generate_assumed_role only creates new role if ASSUMED_ROLE_NAME is not specified

Reply:
comment resolved but problem still exists because the variable ASSUMED_ROLE_NAME is being reused. method will delete it irrespective of the role created by script or user specified it

Reply:
I assumed that no-one would provide this role ARN. Our integration tests are capable of doing it every time, and it only takes a few seconds. Do we want people to have to worry about this field?

Reply:
It needs the privileges to be assumed by the OIDC role, so you would need both of those set up too?

Reply:
yes please, lets create only if not provided by user and delete only if created by the automation

please add this variable to env.example file